packages/react-dom/src/__tests__/ReactTestUtilsAct-test.js

```diff
@@ -246,6 +246,33 @@ function runActTests(label, render, unmount, rerender) {
         ]);
       });
 
+      // @gate __DEV__
+      it('warns if a setState is called outside of act(...) after a component threw', () => {
+        let setValue = null;
+        function App({defaultValue}) {
+          if (defaultValue === undefined) {
+            throw new Error();
+          }
+          const [value, _setValue] = React.useState(defaultValue);
+          setValue = _setValue;
+          return value;
+        }
+
+        expect(() => {
+          act(() => {
+            render(<App defaultValue={undefined} />, container);
+          });
+        }).toThrow();
```

Comment:
Nit: Add an expected message to `toThrow(...)` assertion

Reply:
done: https://github.com/facebook/react/pull/21766/commits/43699a68ba9b9e8d9854aac5e7687163ecb0bd34